api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl.java

```diff
@@ -0,0 +1,288 @@
+import static org.hibernate.criterion.Restrictions.le;
+import static org.hibernate.criterion.Restrictions.not;
+import static org.hibernate.criterion.Subqueries.propertyIn;
+
+import javax.annotation.Nonnull;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.Period;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalAmount;
+import java.time.temporal.TemporalUnit;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.ParamPrefixEnum;
+import ca.uhn.fhir.rest.param.QuantityAndListParam;
+import ca.uhn.fhir.rest.param.QuantityParam;
+import ca.uhn.fhir.rest.param.ReferenceAndListParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hibernate.Criteria;
+import org.hibernate.SessionFactory;
+import org.hibernate.criterion.Criterion;
+import org.hibernate.criterion.DetachedCriteria;
+import org.hibernate.criterion.Projections;
+import org.openmrs.Obs;
+import org.openmrs.Retireable;
+import org.openmrs.Voidable;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.dao.FhirConditionDao;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.openmrs.module.fhir2.api.util.LocalDateTimeFactory;
+import org.openmrs.util.PrivilegeConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PUBLIC)
+@OpenmrsProfile(openmrsPlatformVersion = "2.0.5 - 2.1.*")
+public class FhirConditionDaoImpl extends BaseFhirDao<Obs> implements FhirConditionDao<Obs> {
+	
+	@Qualifier("sessionFactory")
+	@Autowired
+	private SessionFactory sessionFactory;
+	
+	private final boolean isRetireable;
+	
+	private final boolean isVoidable;
+	
+	protected FhirConditionDaoImpl() {
+		this.isRetireable = Retireable.class.isAssignableFrom(typeToken.getRawType());
+		this.isVoidable = Voidable.class.isAssignableFrom(typeToken.getRawType());
+	};
+	
+	private LocalDateTimeFactory localDateTimeFactory;
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public Obs get(@Nonnull String uuid) {
+		return super.get(uuid);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.EDIT_OBS)
+	public Obs createOrUpdate(@Nonnull Obs newEntry) {
+		return super.createOrUpdate(newEntry);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.DELETE_OBS)
+	public Obs delete(@Nonnull String uuid) {
+		return super.delete(uuid);
+	}
+	
+	@Override
+	@Authorized(PrivilegeConstants.GET_OBS)
+	public List<String> getSearchResultUuids(@Nonnull SearchParameterMap theParams) {
```

Comment:
Generally, we don't want to override `getSearchResultsUuids()` and change the implementation. The right way to do this is to add the custom logic we need to the `setupSearchParams()` method (which is overridable for just this purpose). That way we only keep this logic in one place.